src/components/hooks/useUserProfile.ts

```diff
@@ -0,0 +1,14 @@
+import { useAPIContext } from 'components/data/apiContext';
+import { useFetchableData } from 'components/hooks';
+import { UserProfile } from 'models';
+
+/** State hook that returns the user information if logged in, null otherwise */
+export function useUserProfile() {
+    const { getUserProfile } = useAPIContext();
```

Comment:
Does it make this hook somewhat large to be including the values in `useAPIContext`, which had a bunch of spread API objects? Just wondering if not atomizing it has any effect on perf, etc. (I have no evidence to support that it does; just realized that this one function was calling another that has larger scope)

Reply:
Good question! This hook is consuming the value and not returning it. So I'm not concerned about the size. In most cases, it should be consuming the same value that it saw last time.

The broader question is still relevant: the API Context is now constructed by spreading the API functions and then adding the `loginStatus`. Until now, we were setting its default value once and then the Provider would just continue returning that.

Now, the value will be recreated each time the `App` component re-renders. However, that shouldn't be very often, because the only state it depends on that can change is the `loginStatus`.